File: ios/Classes/Bridges/TriggerResultBridge.swift
import Flutter
import SuperwallKit

extension TriggerResult {
  func toJson() -> [String: Any] {
    var dict = [String: Any]()
    switch self {
      case .eventNotFound:
        dict["type"] = "eventNotFound"
      case .noRuleMatch:
        dict["type"] = "noRuleMatch"
      case .paywall(let experiment):
        dict["type"] = "paywall"
        dict["experiment"] = experiment.toJson()
      case .holdout(let experiment):
        dict["type"] = "holdout"
        dict["experiment"] = experiment.toJson()
      case .error(let errorString):
        dict["type"] = "error"
        dict["errorDetail"] = errorString
    }
    return dict
  }
}

File: android/src/main/kotlin/com/superwall/superwallkit_flutter/bridges/TriggerResultBridge.kt
package com.superwall.superwallkit_flutter.bridges

import com.superwall.sdk.models.triggers.TriggerResult

fun TriggerResult.toJson(): Map<String, Any?> {
    return when (this) {
        is TriggerResult.EventNotFound -> mapOf("type" to "eventNotFound")
        is TriggerResult.NoRuleMatch -> mapOf("type" to "noRuleMatch")
        is TriggerResult.Paywall -> mapOf("type" to "paywall", "experiment" to experiment.toJson())
        is TriggerResult.Holdout -> mapOf("type" to "holdout", "experiment" to experiment.toJson())
        is TriggerResult.Error -> mapOf("type" to "error", "errorDetail" to error)
    }
}